cloto/manager/AuthorizationManager.py

```diff
@@ -36,47 +39,96 @@ class AuthorizationManager():
 
-    def checkToken(self, admin_token, token, tenant_id, url):
+        except (KeystoneClientException, KeystoneConnectionRefused) as e:
+            logger.error("No auth token (%s)", e.message)
+            print e.message
+            raise(e)
+
+        return auth_token
+
+    def checkToken(self, admin_token, token, tenant_id, url, auth_api):
         """checks if a token is valid against a url using an admin token."""
-        print("Starting Authentication of token %s " % token)
+        logger.info("Starting Authentication of token %s " % token)
         try:
             if not token:
                 raise Unauthorized("Token is empty")
-            auth_result = self.get_info_token(url, admin_token, token)
+            auth_result = self.get_info_token(url, admin_token, token, auth_api)
             if auth_result:
                 if tenant_id == auth_result.tenant["id"]:
-                    print('The token is valid')
+                    logger.info('The token is valid')
                 else:
+                    logger.error("TenantId %s " % tenant_id)
+                    logger.error("Token TenantId %s " % auth_result.tenant["id"])
                     raise Unauthorized("Token is not valid for specified tenant %s" % tenant_id)
         except Unauthorized as unauth:
+            logger.error(unauth)
             raise unauth
         except InternalServerError as internalError:
             raise AuthorizationFailure("Token could not have enough permissions to access tenant: %s" % tenant_id)
         except Exception as ex:
             raise ex
 
-    def get_info_token(self, url, admin_token, token):
-        headers = {ACCEPT_HEADER: JSON_TYPE, X_AUTH_TOKEN_HEADER: admin_token}
-        r = self.client.get(url + "/" + TOKENS_PATH + token, headers=headers)
-        response = r.text.decode()
-        if response == TOKEN_NOT_FOUND:
-            raise AuthorizationFailure(response)
-        if response == SERVICE_NOT_AUTORIZED:
-            raise AuthorizationFailure("System has an authorization problem, please ask to administrators.")
-        info = json.loads(response)
-        my_token = TokenModel()
-        my_token.expires = info["access"]["token"]["expires"]
-        my_token.id = info["access"]["token"]["id"]
-        my_token.tenant = info["access"]["token"]["tenant"]
-        return my_token
+    def get_info_token(self, url, admin_token, token, auth_api):
+
+        if auth_api == 'v2.0':
+            headers = {ACCEPT_HEADER: JSON_TYPE, X_AUTH_TOKEN_HEADER: admin_token}
+            r = self.client.get(url + "/" + TOKENS_PATH + token, headers=headers)
+
+            if r.status_code != 200 or r.text == "User token not found" or r.text == "Service not authorized":
+                raise AuthorizationFailure(r.text)
+            response = r.text.decode()
+            info = json.loads(response)
+            my_token = TokenModel()
+            my_token.expires = info["access"]["token"]["expires"]
```

Comment:
I should do the follofing here:
tmp = info["access"]["token"]
my_token.expires = tmp["expires"]
my_token.id = tmp["id"]
my_token.tenant = tmp["tenant"]